src/helm/proxy/clients/openai_client.py

```diff
@@ -191,8 +193,83 @@ def do_it():
             embedding=embedding,
         )
 
+    @staticmethod
+    def _get_model_engine_name(model_engine: str) -> str:
+        return "/".join(model_engine.split("/")[1:])
+
     def tokenize(self, request: TokenizationRequest) -> TokenizationRequestResult:
-        raise NotImplementedError("Use the HuggingFaceClient to tokenize.")
+        # For reproducibility purposes, we use huggingface/gpt2 as the default tokenizer.
+        # except for gpt-3.5 turbo models and gpt-4 that use the tiktoken library.
+
+        if (
+            request.model_engine is not None
+            and self._get_model_engine_name(request.model_engine) not in OpenAIClient.MODELS_USING_TIKTOKEN
+        ):
+            return self.tokenizer_client.tokenize(
```

Comment:
Just raise a `ValueError` here rather than forwarding it to another client. This should allow us to remove `self.tokenizer_client` completely. Put a helpful error message in the exception, telling the user to use `huggingface/gpt2` instead.